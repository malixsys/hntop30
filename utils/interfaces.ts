export interface IUnsortedStory {
    id: number;
    title: string;
    loading: boolean;
    commenters: ICommenterCount;
}

export interface IReference {
    once(
        eventType: string,
        successCallback?: (a: IFirebase, b?: string) => any,
        failureCallbackOrContext?: Object | null,
        context?: Object | null
    ): Promise<any>;
}

export interface IFirebase extends IReference {
    child(path: string): IFirebase;

    exists(): boolean;

    exportVal(): any;

    forEach(action: (a: IFirebase) => boolean): boolean;

    getPriority(): string | number | null;

    hasChild(path: string): boolean;

    hasChildren(): boolean;

    key: string | null;

    numChildren(): number;

    ref: IReference;

    toJSON(): Object | null;

    val(): any;
}

export interface IGetItem {
    firebase: IFirebase;
    id: number;
}

export interface IFirebaseStory {
    title: string;
    id: number;
    kids: number[];
}

export interface ILoadedStory {
    title: string;
    id: number;
    kids: ILoadedComment[];
}

export interface ICommenter {
    name: string;
    count: number;
}

export interface IUIStory {
    title: string;
    url?: string;
    id?: number;
    commenters: ICommenter[];
    loading: boolean;
}

export interface ICommenterCount {
    [index: string]: number;
}

export interface IFirebaseComment {
    by?: string;
    kids: number[];
    deleted?: boolean;
}

export interface ILoadedComment {
    by?: string;
    kids: ILoadedComment[];
    deleted?: boolean;
}
