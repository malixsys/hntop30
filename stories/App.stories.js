import React from 'react';
import {App} from "../components/App";

export default {
  title: 'App',
  component: App,
};

export const AppExample = () => (
    <App><h3>children</h3></App>
);

AppExample.story = {
  name: 'Simple App',
};
