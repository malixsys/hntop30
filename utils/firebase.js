/**
 * Recursively load an array of kids by id
 * @param {object} firebase
 * @param {number[]} kids
 * @returns {Promise} an array of kids
 */
const loadKids = async (firebase, kids) => {
  const reads = (kids || []).map(
    async id =>
      new Promise(async resolve => {
        const item = await getItem(firebase, id);
        if(!item) {
            return {}
        }
        const kids = item.kids && item.kids.length > 0 ? await loadKids(firebase, item.kids) : [];
        resolve({ ...item, kids });
      })
  );
  return Promise.all(reads);
};

/**
 * Get an item from Firebase by id
 * @param {object} firebase
 * @param {number} id
 * @returns {Promise<{object}>}
 */
const getItem = (firebase, id) =>
  new Promise(async resolve => {
    if (!id) {
      return resolve({});
    }
    const itemRef = firebase.child('item').child(id);
    itemRef.once('value', async itemSnapshot => {
      const value = itemSnapshot.val();
      resolve(value);
    });
  });

/**
 * read a story from Firebase by id
 * @param {object} firebase
 * @param {number} id
 * @returns {Promise<{object}>}
 */
export const readStory = (firebase, id) =>
  new Promise(async resolve => {
    const story = await getItem(firebase, id);
    const kids = story.kids && story.kids.length > 0 ? await loadKids(firebase, story.kids) : [];
    resolve({ ...story, kids });
  });
