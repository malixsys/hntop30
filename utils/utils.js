const MAX_commenters = 10;

const findcommenters = (comments, commenters = {}) => {
  (comments || []).forEach(comment => {
    const { by, kids, deleted } = comment;
    if (!(!by || deleted)) {
      if (!commenters[by]) {
        commenters[by] = 0;
      }
      commenters[by] += 1;
      findcommenters(kids, commenters);
    }
  });
};

const bestCommentor = (a, b) => {
  const [, countA] = a;
  const [, countB] = b;
  if (countA > countB) {
    return -1;
  } else if (countA < countB) {
    return 1;
  } else {
    return 0;
  }
};

const sortResults = ({ commenters, ...rest }) => {
  const entries = Object.entries(commenters);
  entries.sort(bestCommentor);
  return {
    ...rest,
    commenters: entries.map(([name, count]) => ({ name, count })).filter((_, i) => i < MAX_commenters)
  };
};

export const processStories = stories => {
  const results = stories.map(({ kids, ...rest }) => {
    const commenters = {};
    findcommenters(kids, commenters);
    return { ...rest, commenters };
  });
  return results.map(sortResults);
};

export const countComments = (stories, name) => {
  return stories.reduce((m, story) => {
    story.commenters.forEach(c => {
      if (c.name === name) {
        m += c.count;
      }
    });
    return m;
  }, 0);
};
