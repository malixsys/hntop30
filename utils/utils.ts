import {ICommenterCount, ILoadedComment, ILoadedStory, IUIStory, IUnsortedStory} from "./interfaces";

const MAX_commenters = 10;

const findCommenters = (comments: ILoadedComment[], commenters: ICommenterCount = {}) => {
  (comments || []).forEach(comment => {
    const { by, kids, deleted } = comment;
    if (!(!by || deleted)) {
      if (!commenters[by]) {
        commenters[by] = 0;
      }
      commenters[by] += 1;
      findCommenters(kids, commenters);
    }
  });
};

const bestCommentor = (a: [any, number], b: [any, number]) => {
  const [, countA] = a;
  const [, countB] = b;
  if (countA > countB) {
    return -1;
  } else if (countA < countB) {
    return 1;
  } else {
    return 0;
  }
};

const sortResults: ({ commenters, ...rest }: IUnsortedStory) => IUIStory = ({
  commenters,
  ...rest
}: IUnsortedStory) => {
  const entries: [string, number][] = Object.entries(commenters);
  entries.sort(bestCommentor);
  return {
    ...rest,
    commenters: entries.map(([name, count]) => ({ name, count })).filter((_, i) => i < MAX_commenters)
  };
};

export const processStories: (stories: ILoadedStory[]) => IUIStory[] = (stories: ILoadedStory[]) => {
  const results: IUnsortedStory[] = stories.map(({ kids, ...rest }) => {
    const commenters: ICommenterCount = {};
    findCommenters(kids, commenters);
    return { ...rest, commenters, loading: false };
  });
  return results.map(sortResults);
};

export const countComments = (stories: { commenters: { name: string; count: number }[] }[], name: string) => {
  return stories.reduce((m, story) => {
    story.commenters.forEach(c => {
      if (c.name === name) {
        m += c.count;
      }
    });
    return m;
  }, 0);
};

