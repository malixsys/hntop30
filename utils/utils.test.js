import stories from '../test/top.json';
import smallStories from '../test/small.json';
import processedStories from '../test/stories.json';
import { processStories, countComments } from './utils';

describe('countComments', () => {
  it('returns proper values', () => {
    expect(countComments(processedStories, 'user-a')).toEqual(9);
    expect(countComments(processedStories, 'user-b')).toEqual(10);
    expect(countComments(processedStories, 'user-c')).toEqual(8);
  });
});

describe('processStories', () => {
  it('returns proper values for small result', () => {
    const result = processStories(smallStories);
    expect(result).toEqual(processedStories);
  });
  it('returns proper values for big result', () => {
    const result = processStories(stories);
    expect(result).toMatchSnapshot();
  });
});
