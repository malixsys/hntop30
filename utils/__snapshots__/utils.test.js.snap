// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`processStories returns proper values for big result 1`] = `
Array [
  Object {
    "by": "robert-boehnke",
    "commenters": Array [
      Object {
        "count": 3,
        "name": "kragen",
      },
      Object {
        "count": 3,
        "name": "pier25",
      },
      Object {
        "count": 2,
        "name": "anderspitman",
      },
      Object {
        "count": 2,
        "name": "TeMPOraL",
      },
      Object {
        "count": 2,
        "name": "amitp",
      },
      Object {
        "count": 2,
        "name": "nimish",
      },
      Object {
        "count": 2,
        "name": "gugagore",
      },
      Object {
        "count": 2,
        "name": "andonisus",
      },
      Object {
        "count": 2,
        "name": "jmiskovic",
      },
      Object {
        "count": 2,
        "name": "Sohcahtoa82",
      },
    ],
    "descendants": 104,
    "id": 22310813,
    "score": 1206,
    "time": 1581529100,
    "title": "Gears",
    "type": "story",
    "url": "https://ciechanow.ski/gears/",
  },
  Object {
    "by": "twapi",
    "commenters": Array [
      Object {
        "count": 6,
        "name": "billygriffin22",
      },
      Object {
        "count": 3,
        "name": "geerlingguy",
      },
      Object {
        "count": 3,
        "name": "OJFord",
      },
      Object {
        "count": 2,
        "name": "SEJeff",
      },
      Object {
        "count": 2,
        "name": "eeZah7Ux",
      },
      Object {
        "count": 2,
        "name": "Spiritus",
      },
      Object {
        "count": 2,
        "name": "vmchale",
      },
      Object {
        "count": 2,
        "name": "guessmyname",
      },
      Object {
        "count": 2,
        "name": "scarejunba",
      },
      Object {
        "count": 2,
        "name": "tilolebo",
      },
    ],
    "descendants": 98,
    "id": 22310516,
    "score": 352,
    "time": 1581527145,
    "title": "GitHub CLI is now in beta",
    "type": "story",
    "url": "https://github.blog/2020-02-12-supercharge-your-command-line-experience-github-cli-is-now-in-beta/",
  },
  Object {
    "by": "kick",
    "commenters": Array [
      Object {
        "count": 2,
        "name": "PaulHoule",
      },
      Object {
        "count": 1,
        "name": "cellularmitosis",
      },
      Object {
        "count": 1,
        "name": "dmitripopov",
      },
      Object {
        "count": 1,
        "name": "downerending",
      },
      Object {
        "count": 1,
        "name": "glic3rinu",
      },
      Object {
        "count": 1,
        "name": "john-radio",
      },
      Object {
        "count": 1,
        "name": "ww520",
      },
      Object {
        "count": 1,
        "name": "kick",
      },
    ],
    "descendants": 9,
    "id": 22313003,
    "score": 36,
    "time": 1581543018,
    "title": "A Bestiary of Single-File Implementations of Programming Languages",
    "type": "story",
    "url": "https://github.com/marcpaq/b1fipl",
  },
  Object {
    "by": "rhaidiz",
    "commenters": Array [
      Object {
        "count": 2,
        "name": "big_chungus",
      },
      Object {
        "count": 1,
        "name": "_wldu",
      },
      Object {
        "count": 1,
        "name": "rhaidiz",
      },
    ],
    "descendants": 4,
    "id": 22313131,
    "score": 24,
    "time": 1581543843,
    "title": "Broxy: An intercept proxy written in Go",
    "type": "story",
    "url": "https://github.com/rhaidiz/broxy",
  },
  Object {
    "by": "yarapavan",
    "commenters": Array [
      Object {
        "count": 1,
        "name": "dvt",
      },
      Object {
        "count": 1,
        "name": "GaryNumanVevo",
      },
      Object {
        "count": 1,
        "name": "morley",
      },
      Object {
        "count": 1,
        "name": "newprint",
      },
      Object {
        "count": 1,
        "name": "kempbellt",
      },
      Object {
        "count": 1,
        "name": "Dayjagupta",
      },
      Object {
        "count": 1,
        "name": "oh_sigh",
      },
    ],
    "descendants": 7,
    "id": 22297998,
    "score": 93,
    "time": 1581425074,
    "title": "An Astronaut’s Guide to Mental Models",
    "type": "story",
    "url": "https://fs.blog/2020/02/mental-models-in-space/",
  },
  Object {
    "by": "minimaxir",
    "commenters": Array [
      Object {
        "count": 5,
        "name": "tempsy",
      },
      Object {
        "count": 4,
        "name": "coldpie",
      },
      Object {
        "count": 4,
        "name": "notyourday",
      },
      Object {
        "count": 4,
        "name": "dannyw",
      },
      Object {
        "count": 3,
        "name": "dang",
      },
      Object {
        "count": 3,
        "name": "bdcravens",
      },
      Object {
        "count": 3,
        "name": "swalsh",
      },
      Object {
        "count": 2,
        "name": "jolmg",
      },
      Object {
        "count": 2,
        "name": "wlesieutre",
      },
      Object {
        "count": 2,
        "name": "jonas21",
      },
    ],
    "descendants": 155,
    "id": 22311126,
    "score": 143,
    "time": 1581531080,
    "title": "Essential shutting down after failing to launch ‘GEM’ smartphone",
    "type": "story",
    "url": "https://9to5google.com/2020/02/12/essential-shutting-down/",
  },
  Object {
    "by": "theomega",
    "commenters": Array [
      Object {
        "count": 5,
        "name": "mojuba",
      },
      Object {
        "count": 4,
        "name": "myself248",
      },
      Object {
        "count": 3,
        "name": "vonseel",
      },
      Object {
        "count": 3,
        "name": "goatherders",
      },
      Object {
        "count": 3,
        "name": "dzhiurgis",
      },
      Object {
        "count": 3,
        "name": "muglug",
      },
      Object {
        "count": 3,
        "name": "zippergz",
      },
      Object {
        "count": 3,
        "name": "seppin",
      },
      Object {
        "count": 3,
        "name": "RandomBacon",
      },
      Object {
        "count": 3,
        "name": "UI_at_80x24",
      },
    ],
    "descendants": 179,
    "id": 22308211,
    "score": 299,
    "time": 1581510100,
    "title": "SoundCloud raises $75M from SiriusXM",
    "type": "story",
    "url": "https://techcrunch.com/2020/02/11/music-streaming-pioneer-soundcloud-raises-75m-from-pandora-owner-siriusxm",
  },
  Object {
    "by": "SZJX",
    "commenters": Array [
      Object {
        "count": 4,
        "name": "petrogradphilos",
      },
      Object {
        "count": 4,
        "name": "dooglius",
      },
      Object {
        "count": 4,
        "name": "stef25",
      },
      Object {
        "count": 4,
        "name": "andai",
      },
      Object {
        "count": 4,
        "name": "K0SM0S",
      },
      Object {
        "count": 4,
        "name": "garmaine",
      },
      Object {
        "count": 3,
        "name": "bildung",
      },
      Object {
        "count": 3,
        "name": "dotancohen",
      },
      Object {
        "count": 3,
        "name": "vintermann",
      },
      Object {
        "count": 3,
        "name": "harryh",
      },
    ],
    "descendants": 186,
    "id": 22307445,
    "score": 486,
    "time": 1581501006,
    "title": "No person who was born blind has ever been diagnosed with schizophrenia",
    "type": "story",
    "url": "https://www.vice.com/en_us/article/939qbz/people-born-blind-are-mysteriously-protected-from-schizophrenia",
  },
  Object {
    "by": "seleniumbase",
    "commenters": Array [
      Object {
        "count": 2,
        "name": "Afton",
      },
      Object {
        "count": 1,
        "name": "gnagatomo",
      },
      Object {
        "count": 1,
        "name": "sbergot",
      },
      Object {
        "count": 1,
        "name": "swampthing",
      },
      Object {
        "count": 1,
        "name": "luhego",
      },
      Object {
        "count": 1,
        "name": "ratbeard",
      },
      Object {
        "count": 1,
        "name": "bluntfang",
      },
      Object {
        "count": 1,
        "name": "misiti3780",
      },
    ],
    "descendants": 9,
    "id": 22297812,
    "score": 28,
    "time": 1581422717,
    "title": "Using Python and Selenium for Automated Visual Regression Testing",
    "type": "story",
    "url": "https://github.com/seleniumbase/SeleniumBase/blob/master/examples/visual_testing/ReadMe.md",
  },
  Object {
    "by": "ajaviaad",
    "commenters": Array [
      Object {
        "count": 2,
        "name": "eindiran",
      },
      Object {
        "count": 1,
        "name": "bllguo",
      },
      Object {
        "count": 1,
        "name": "slim",
      },
      Object {
        "count": 1,
        "name": "aldoushuxley001",
      },
      Object {
        "count": 1,
        "name": "ConsiderCrying",
      },
      Object {
        "count": 1,
        "name": "jborichevskiy",
      },
      Object {
        "count": 1,
        "name": "andrewflnr",
      },
      Object {
        "count": 1,
        "name": "JTechno",
      },
      Object {
        "count": 1,
        "name": "EamonnMR",
      },
      Object {
        "count": 1,
        "name": "beepboopbeep",
      },
    ],
    "descendants": 11,
    "id": 22312864,
    "score": 25,
    "time": 1581541956,
    "title": "Hallucinogenic Fish",
    "type": "story",
    "url": "https://en.wikipedia.org/wiki/Hallucinogenic_fish",
  },
  Object {
    "by": "coloneltcb",
    "commenters": Array [
      Object {
        "count": 5,
        "name": "downerending",
      },
      Object {
        "count": 4,
        "name": "merqurio",
      },
      Object {
        "count": 4,
        "name": "ghaff",
      },
      Object {
        "count": 3,
        "name": "retsibsi",
      },
      Object {
        "count": 3,
        "name": "baddox",
      },
      Object {
        "count": 2,
        "name": "Filligree",
      },
      Object {
        "count": 2,
        "name": "gamblor956",
      },
      Object {
        "count": 2,
        "name": "berkeleyjunk",
      },
      Object {
        "count": 2,
        "name": "kalipsosu",
      },
      Object {
        "count": 2,
        "name": "maa5444",
      },
    ],
    "descendants": 78,
    "id": 22311618,
    "score": 76,
    "time": 1581533891,
    "title": "MWC 2020 canceled over coronavirus health concerns",
    "type": "story",
    "url": "https://www.theverge.com/2020/2/12/21127754/mwc-2020-canceled-coronavirus-trade-show-phone-mobile-world-congress-gsma-statement",
  },
  Object {
    "by": "keithwhor",
    "commenters": Array [
      Object {
        "count": 18,
        "name": "keithwhor",
      },
      Object {
        "count": 2,
        "name": "samblr",
      },
      Object {
        "count": 2,
        "name": "brundolf",
      },
      Object {
        "count": 1,
        "name": "crabasa",
      },
      Object {
        "count": 1,
        "name": "avip",
      },
      Object {
        "count": 1,
        "name": "sarakayakomzin",
      },
      Object {
        "count": 1,
        "name": "zerkten",
      },
      Object {
        "count": 1,
        "name": "SahAssar",
      },
      Object {
        "count": 1,
        "name": "ryanianian",
      },
      Object {
        "count": 1,
        "name": "howmayiannoyyou",
      },
    ],
    "descendants": 42,
    "id": 22306996,
    "score": 121,
    "time": 1581496140,
    "title": "Show HN: Autocode – Automatically generate API code for SaaS apps",
    "type": "story",
    "url": "https://autocode.com/",
  },
  Object {
    "by": "mroche",
    "commenters": Array [
      Object {
        "count": 2,
        "name": "jhasse",
      },
      Object {
        "count": 1,
        "name": "unixhero",
      },
      Object {
        "count": 1,
        "name": "loeg",
      },
      Object {
        "count": 1,
        "name": "romaniitedomum",
      },
      Object {
        "count": 1,
        "name": "inetknght",
      },
      Object {
        "count": 1,
        "name": "qeqeqeqe",
      },
    ],
    "descendants": 7,
    "id": 22310793,
    "score": 57,
    "time": 1581528970,
    "title": "Toward _FORTIFY_SOURCE Parity Between Clang and GCC",
    "type": "story",
    "url": "https://developers.redhat.com/blog/2020/02/11/toward-_fortify_source-parity-between-clang-and-gcc/",
  },
  Object {
    "by": "adrelanos",
    "commenters": Array [
      Object {
        "count": 2,
        "name": "wyldfire",
      },
      Object {
        "count": 2,
        "name": "Hello71",
      },
      Object {
        "count": 2,
        "name": "CameronNemo",
      },
      Object {
        "count": 1,
        "name": "LinuxBender",
      },
      Object {
        "count": 1,
        "name": "jws",
      },
      Object {
        "count": 1,
        "name": "heavenlyblue",
      },
      Object {
        "count": 1,
        "name": "kevingadd",
      },
      Object {
        "count": 1,
        "name": "swatkat",
      },
      Object {
        "count": 1,
        "name": "TrueDuality",
      },
      Object {
        "count": 1,
        "name": "cbsks",
      },
    ],
    "descendants": 25,
    "id": 22308882,
    "score": 151,
    "time": 1581516331,
    "title": "Linux Kernel Runtime Guard",
    "type": "story",
    "url": "https://www.whonix.org/wiki/Linux_Kernel_Runtime_Guard_LKRG",
  },
  Object {
    "by": "nallerooth",
    "commenters": Array [
      Object {
        "count": 10,
        "name": "DonHopkins",
      },
      Object {
        "count": 6,
        "name": "JdeBP",
      },
      Object {
        "count": 5,
        "name": "krilovsky",
      },
      Object {
        "count": 4,
        "name": "jamiethompson",
      },
      Object {
        "count": 3,
        "name": "_-___________-_",
      },
      Object {
        "count": 3,
        "name": "pratio",
      },
      Object {
        "count": 3,
        "name": "taneq",
      },
      Object {
        "count": 3,
        "name": "walrus01",
      },
      Object {
        "count": 3,
        "name": "croon",
      },
      Object {
        "count": 3,
        "name": "C1sc0cat",
      },
    ],
    "descendants": 145,
    "id": 22306801,
    "score": 468,
    "time": 1581494045,
    "title": "Rotary Cellphone",
    "type": "story",
    "url": "http://justine-haupt.com/rotarycellphone/",
  },
  Object {
    "by": "louwrentius",
    "commenters": Array [
      Object {
        "count": 4,
        "name": "ramraj07",
      },
      Object {
        "count": 3,
        "name": "threeseed",
      },
      Object {
        "count": 3,
        "name": "Dylan16807",
      },
      Object {
        "count": 3,
        "name": "randyzwitch",
      },
      Object {
        "count": 2,
        "name": "dijit",
      },
      Object {
        "count": 2,
        "name": "dmos62",
      },
      Object {
        "count": 2,
        "name": "lotyrin",
      },
      Object {
        "count": 2,
        "name": "keanzu",
      },
      Object {
        "count": 2,
        "name": "takeda",
      },
      Object {
        "count": 2,
        "name": "chmod775",
      },
    ],
    "descendants": 86,
    "id": 22309883,
    "score": 117,
    "time": 1581523302,
    "title": "Does my data fit in RAM?",
    "type": "story",
    "url": "https://yourdatafitsinram.net/",
  },
  Object {
    "by": "shawkinaw",
    "commenters": Array [
      Object {
        "count": 2,
        "name": "bgentry",
      },
      Object {
        "count": 2,
        "name": "mrob",
      },
      Object {
        "count": 1,
        "name": "frandroid",
      },
      Object {
        "count": 1,
        "name": "rendall",
      },
      Object {
        "count": 1,
        "name": "kevinoid",
      },
      Object {
        "count": 1,
        "name": "scrooched_moose",
      },
      Object {
        "count": 1,
        "name": "ahoy",
      },
      Object {
        "count": 1,
        "name": "cavanasm",
      },
      Object {
        "count": 1,
        "name": "dhritzkiv",
      },
      Object {
        "count": 1,
        "name": "asdfman123",
      },
    ],
    "descendants": 21,
    "id": 22309100,
    "score": 90,
    "time": 1581518205,
    "title": "Beer Waste Saves Montana Town $1M on Water Treatment",
    "type": "story",
    "url": "https://www.npr.org/2020/02/12/804586191/beer-waste-saves-montana-town-1-million-on-water-treatment",
  },
  Object {
    "by": "LinuxBender",
    "commenters": Array [
      Object {
        "count": 12,
        "name": "troquerre",
      },
      Object {
        "count": 5,
        "name": "RL_Quine",
      },
      Object {
        "count": 5,
        "name": "danShumway",
      },
      Object {
        "count": 5,
        "name": "Ajedi32",
      },
      Object {
        "count": 5,
        "name": "pinhead26",
      },
      Object {
        "count": 4,
        "name": "tynes",
      },
      Object {
        "count": 4,
        "name": "dang",
      },
      Object {
        "count": 4,
        "name": "xoa",
      },
      Object {
        "count": 4,
        "name": "lifty",
      },
      Object {
        "count": 4,
        "name": "xur17",
      },
    ],
    "descendants": 95,
    "id": 22310604,
    "score": 183,
    "time": 1581527734,
    "title": "Decentralized Naming and Certificate Authority",
    "type": "story",
    "url": "https://handshake.org/",
  },
  Object {
    "by": "JonAtkinson",
    "commenters": Array [
      Object {
        "count": 3,
        "name": "mhd",
      },
      Object {
        "count": 3,
        "name": "JulianMorrison",
      },
      Object {
        "count": 3,
        "name": "alxlaz",
      },
      Object {
        "count": 2,
        "name": "qubex",
      },
      Object {
        "count": 2,
        "name": "badsectoracula",
      },
      Object {
        "count": 2,
        "name": "AdmiralAsshat",
      },
      Object {
        "count": 2,
        "name": "Lammy",
      },
      Object {
        "count": 2,
        "name": "uranium235",
      },
      Object {
        "count": 1,
        "name": "favorited",
      },
      Object {
        "count": 1,
        "name": "jsz0",
      },
    ],
    "descendants": 61,
    "id": 22308138,
    "score": 161,
    "time": 1581509272,
    "title": "Nextspace – NeXTSTEP-like desktop environment for Linux",
    "type": "story",
    "url": "https://github.com/trunkmaster/nextspace",
  },
  Object {
    "by": "smacktoward",
    "commenters": Array [
      Object {
        "count": 4,
        "name": "the_af",
      },
      Object {
        "count": 3,
        "name": "mcv",
      },
      Object {
        "count": 2,
        "name": "ConfusedDog",
      },
      Object {
        "count": 2,
        "name": "leftyted",
      },
      Object {
        "count": 2,
        "name": "rehasu",
      },
      Object {
        "count": 2,
        "name": "Trombone12",
      },
      Object {
        "count": 2,
        "name": "speeder",
      },
      Object {
        "count": 1,
        "name": "kazinator",
      },
      Object {
        "count": 1,
        "name": "cbanek",
      },
      Object {
        "count": 1,
        "name": "Seenso",
      },
    ],
    "descendants": 48,
    "id": 22296312,
    "score": 79,
    "time": 1581398885,
    "title": "Who Was Sun Tzu’s Napoleon?",
    "type": "story",
    "url": "https://thestrategybridge.org/the-bridge/2020/2/11/who-was-sun-tzus-napoleon",
  },
  Object {
    "by": "DoreenMichele",
    "commenters": Array [
      Object {
        "count": 11,
        "name": "bluGill",
      },
      Object {
        "count": 8,
        "name": "rayiner",
      },
      Object {
        "count": 7,
        "name": "kop316",
      },
      Object {
        "count": 6,
        "name": "lotsofpulp",
      },
      Object {
        "count": 4,
        "name": "atombender",
      },
      Object {
        "count": 4,
        "name": "ghaff",
      },
      Object {
        "count": 4,
        "name": "reggieband",
      },
      Object {
        "count": 3,
        "name": "PaulDavisThe1st",
      },
      Object {
        "count": 3,
        "name": "dsr_",
      },
      Object {
        "count": 3,
        "name": "mc32",
      },
    ],
    "descendants": 169,
    "id": 22296477,
    "score": 220,
    "time": 1581401847,
    "title": "The Rise of “Movie-Set Urbanism”",
    "type": "story",
    "url": "https://www.strongtowns.org/journal/2020/2/6/movie-set-urbanism",
  },
  Object {
    "by": "sohkamyung",
    "commenters": Array [
      Object {
        "count": 1,
        "name": "microtherion",
      },
      Object {
        "count": 1,
        "name": "compiler-guy",
      },
    ],
    "descendants": 2,
    "id": 22297212,
    "score": 64,
    "time": 1581413748,
    "title": "A modern take on the “paperclip computer”",
    "type": "story",
    "url": "https://hackaday.com/2020/02/08/a-modern-take-on-the-paperclip-computer/",
  },
  Object {
    "by": "i_am_not_elon",
    "commenters": Array [
      Object {
        "count": 1,
        "name": "protomyth",
      },
      Object {
        "count": 1,
        "name": "sturgill",
      },
      Object {
        "count": 1,
        "name": "bryanrasmussen",
      },
      Object {
        "count": 1,
        "name": "lifeisstillgood",
      },
      Object {
        "count": 1,
        "name": "strbean",
      },
      Object {
        "count": 1,
        "name": "pdonis",
      },
      Object {
        "count": 1,
        "name": "throwaway713",
      },
      Object {
        "count": 1,
        "name": "jka",
      },
      Object {
        "count": 1,
        "name": "beeschlenker",
      },
    ],
    "descendants": 8,
    "id": 22308701,
    "score": 67,
    "time": 1581514801,
    "title": "First, Do No Harm",
    "type": "story",
    "url": "https://stratechery.com/2020/first-do-no-harm/",
  },
  Object {
    "by": "AdriaanvRossum",
    "commenters": Array [
      Object {
        "count": 12,
        "name": "AdriaanvRossum",
      },
      Object {
        "count": 4,
        "name": "smt88",
      },
      Object {
        "count": 3,
        "name": "radicalbyte",
      },
      Object {
        "count": 2,
        "name": "XCSme",
      },
      Object {
        "count": 2,
        "name": "misiti3780",
      },
      Object {
        "count": 2,
        "name": "balls187",
      },
      Object {
        "count": 2,
        "name": "rhacker",
      },
      Object {
        "count": 2,
        "name": "mritchie712",
      },
      Object {
        "count": 2,
        "name": "ronreiter",
      },
      Object {
        "count": 2,
        "name": "mason55",
      },
    ],
    "descendants": 97,
    "id": 22309316,
    "score": 244,
    "time": 1581519853,
    "title": "Simple Analytics hits $4k MRR and shares its numbers",
    "type": "story",
    "url": "https://simpleanalytics.com/open",
  },
  Object {
    "by": "pseudolus",
    "commenters": Array [
      Object {
        "count": 6,
        "name": "dmitryminkovsky",
      },
      Object {
        "count": 5,
        "name": "duxup",
      },
      Object {
        "count": 4,
        "name": "bananabreakfast",
      },
      Object {
        "count": 3,
        "name": "arminiusreturns",
      },
      Object {
        "count": 3,
        "name": "WhompingWindows",
      },
      Object {
        "count": 3,
        "name": "Seenso",
      },
      Object {
        "count": 3,
        "name": "baconcandy",
      },
      Object {
        "count": 2,
        "name": "Gaussian",
      },
      Object {
        "count": 2,
        "name": "mullingitover",
      },
      Object {
        "count": 2,
        "name": "rsync",
      },
    ],
    "descendants": 82,
    "id": 22297677,
    "score": 72,
    "time": 1581420762,
    "title": "Colorado voters are set to decide if wolves should be reintroduced to the state",
    "type": "story",
    "url": "https://www.npr.org/2020/02/11/803895398/colorado-voters-are-set-to-decide-if-wolves-should-be-reintroduced-to-the-state",
  },
  Object {
    "by": "signa11",
    "commenters": Array [
      Object {
        "count": 3,
        "name": "Supermancho",
      },
      Object {
        "count": 3,
        "name": "codingslave",
      },
      Object {
        "count": 2,
        "name": "karatestomp",
      },
      Object {
        "count": 2,
        "name": "rednerrus",
      },
      Object {
        "count": 2,
        "name": "jakevoytko",
      },
      Object {
        "count": 2,
        "name": "buboard",
      },
      Object {
        "count": 2,
        "name": "chasing",
      },
      Object {
        "count": 2,
        "name": "ScottFree",
      },
      Object {
        "count": 2,
        "name": "poulsbohemian",
      },
      Object {
        "count": 1,
        "name": "hyperpallium",
      },
    ],
    "descendants": 47,
    "id": 22309291,
    "score": 62,
    "time": 1581519727,
    "title": "Why are we so bad at software engineering?",
    "type": "story",
    "url": "https://www.bitlog.com/2020/02/12/why-are-we-so-bad-at-software-engineering/",
  },
  Object {
    "by": "IA21",
    "commenters": Array [
      Object {
        "count": 5,
        "name": "nk1tz",
      },
      Object {
        "count": 4,
        "name": "claudiawerner",
      },
      Object {
        "count": 4,
        "name": "2038AD",
      },
      Object {
        "count": 4,
        "name": "simiones",
      },
      Object {
        "count": 3,
        "name": "karatestomp",
      },
      Object {
        "count": 3,
        "name": "yxhuvud",
      },
      Object {
        "count": 3,
        "name": "jmcmichael",
      },
      Object {
        "count": 3,
        "name": "neaden",
      },
      Object {
        "count": 3,
        "name": "jkhdigital",
      },
      Object {
        "count": 3,
        "name": "codingmess",
      },
    ],
    "descendants": 122,
    "id": 22297291,
    "score": 124,
    "time": 1581414871,
    "title": "The Myth of the Barter Economy (2016)",
    "type": "story",
    "url": "https://www.theatlantic.com/business/archive/2016/02/barter-society-myth/471051/",
  },
  Object {
    "by": "ssully",
    "commenters": Array [
      Object {
        "count": 11,
        "name": "kube-system",
      },
      Object {
        "count": 7,
        "name": "CWuestefeld",
      },
      Object {
        "count": 6,
        "name": "dustfinger",
      },
      Object {
        "count": 5,
        "name": "pbhjpbhj",
      },
      Object {
        "count": 5,
        "name": "jessaustin",
      },
      Object {
        "count": 5,
        "name": "dependenttypes",
      },
      Object {
        "count": 4,
        "name": "gruez",
      },
      Object {
        "count": 4,
        "name": "jariel",
      },
      Object {
        "count": 4,
        "name": "owenmarshall",
      },
      Object {
        "count": 3,
        "name": "SAI_Peregrinus",
      },
    ],
    "descendants": 186,
    "id": 22309310,
    "score": 184,
    "time": 1581519830,
    "title": "US says it can prove Huawei has backdoor access to mobile-phone networks",
    "type": "story",
    "url": "https://arstechnica.com/tech-policy/2020/02/us-gave-allies-evidence-that-huawei-can-snoop-on-phone-networks-wsj-says/",
  },
  Object {
    "by": "mindcrime",
    "commenters": Array [],
    "descendants": 0,
    "id": 22313088,
    "score": 3,
    "time": 1581543603,
    "title": "Trusted Automated Exchange of Intelligence Information (Taxii) [pdf]",
    "type": "story",
    "url": "https://docs.oasis-open.org/cti/taxii/v2.1/cs01/taxii-v2.1-cs01.pdf",
  },
  Object {
    "by": "mortinie",
    "commenters": Array [],
    "descendants": 0,
    "id": 22297159,
    "score": 16,
    "time": 1581412972,
    "title": "GAN Collaborative Image Inpainting",
    "type": "story",
    "url": "https://github.com/Mortiniera/collaborative-image-inpainting",
  },
]
`;
